hq/app/logic/SnykDisplay.scala

```diff
@@ -1,81 +1,63 @@
-      org <- orglist.find(_.name == requiredOrganisation.value)
-    } yield org
-    val error = parseJsonToError(s)
-    val f = Failure(s"Unable to find organisation from $s", s"Could not read Snyk response (${error.error})", 502, None, None)
-    Attempt.fromOption(organisation, FailedAttempt(f))
+  def parseOrganisations(rawJson: String, snykGroupId: SnykGroupId)(implicit ec: ExecutionContext): Attempt[List[SnykOrganisation]] = {
+    for {
+      organisationList <- parseJsonToOrganisationList(rawJson)
+      guardianOrganisationList = organisationList filter {
+        case SnykOrganisation(_, _, Some(group)) => group.id==snykGroupId.value
+        case _ => false
+      }
+    } yield guardianOrganisationList
   }
 
-  private def parseJsonToError(s: String): SnykError = try {
-    Json.parse(s).asOpt[SnykError].getOrElse(SnykError(s))
-  } catch {
-    case _: JsonParseException => SnykError(s)
+  private def parseJsonToError(s: String): SnykError = Try(Json.parse(s).validateOpt[SnykError]) match {
+    case Success(JsSuccess(Some(error), _)) => error
+    case _ => SnykError(s)
   }
 
-  private def parseJsonToOrganisationList(s: String): Option[List[SnykOrganisation]] = try {
-    (Json.parse(s) \ "orgs").asOpt[List[SnykOrganisation]]
-  } catch {
-    case _: JsonParseException => None
+  def parseJsonToObject[A](label: String, rawJson: String, f: (String => JsResult[A])): Attempt[A] = Try(f(rawJson)) match {
+    case Success(JsSuccess(parsedObject, _)) => Attempt.Right(parsedObject)
+    case Success(JsError(e)) =>
+      val error = parseJsonToError(rawJson)
+      val failure = Failure(s"Unable to find $label from $rawJson: $e", s"Could not read Snyk response (${error.error})", 502, None, None)
+      Attempt.Left(failure)
+    case scala.util.Failure(e) =>
+      val failure = Failure(s"Unable to find $label from $rawJson", s"Could not read Snyk response ($rawJson)", 502, None, Some(e))
+      Attempt.Left(failure)
   }
 
-  def getProjectIdList(s: String): Attempt[List[SnykProject]] = {
-    val projects = parseJsonToProjectIdList(s)
-    val error = parseJsonToError(s)
-    val f = Failure(s"Unable to find project ids from $s", s"Could not read Snyk response (${error.error})", 502, None, None)
-    Attempt.fromOption(projects, FailedAttempt(f))
-  }
+  def getProjectIdList(organisationAndRequestList: List[(SnykOrganisation, String)])(implicit ec: ExecutionContext): Attempt[List[((SnykOrganisation, String), List[SnykProject])]] =
+    Attempt.labelledTraverse(organisationAndRequestList) { s => parseJsonToProjectIdList(s._2) }
 
-  private def parseJsonToProjectIdList(s: String) = try {
-    (Json.parse(s) \ "projects").asOpt[List[SnykProject]]
-  } catch {
-    case _: JsonParseException => None
-  }
+  def parseJsonToOrganisationList(s: String): Attempt[List[SnykOrganisation]] = parseJsonToObject("organisations", s, body => {(Json.parse(body) \ "orgs").validate[List[SnykOrganisation]]} )
```

Comment:
Very long line (implementation on the next line again). there's also redundant `{  ...  }` around the `(Json.parse ... )` bit. Space before the closing `)` looks out of place as well.

Reply:
*May* be clearer with `_` instead of `body => ... body ...`, see what you think!